sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/TransactionCoordinator.java

```diff
@@ -101,4 +105,20 @@
                 }
             });
     }
+
+    @Override
+    public void dispose() {
+        if (isDisposed.get()) return;
+
+        isDisposed.set(true);
```

Comment:
This doesn't really make it atomic. Multiple threads could still reach this point concurrently.